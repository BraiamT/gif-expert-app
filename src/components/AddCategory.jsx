import { useState } from 'react';
import { PropTypes } from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // setCategories(categories => [ inputValue, ...categories ]); // el setXyz ya trae el objeto
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="input-form">
            <input
                type="text"
                placeholder="Buscar GIFs"
                value={ inputValue }
                // onChange={ (event) => onInputChange(event) }
                // No es necesario mandar aquÃ­ el evento porque siempre se recibe por defecto
                onChange={ onInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
