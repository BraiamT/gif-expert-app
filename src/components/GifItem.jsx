import PropTypes from 'prop-types';

export const GifItem = ({ title, imgUrl }) => {

    return (
        <div className="card">
            <img src={ imgUrl } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    imgUrl: PropTypes.string.isRequired
}
