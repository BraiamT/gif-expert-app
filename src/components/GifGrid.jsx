import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { PropTypes } from 'prop-types';

export const GifGrid = ({ category }) => {

    /* CUSTOM HOOK useFetchGifs */
    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>
            {
                // isLoading ? ( <h2>Loading...</h2> ) : null // null no se renderiza en React
                isLoading && ( <h2>Loading...</h2> )
            }
            {
                ( images.length === 0 && !isLoading ) && ( <h2>No Results...</h2> )
            }

            <div className='card-grid'>
            {
                images.map( (image) =>
                    (
                        <GifItem
                            key={ image.id }
                            { ...image } // Esto esparce las propiedades de image al componente en cuestiÃ³n
                        />
                    )
                )
            }
            </div>
        </>
    );

}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
