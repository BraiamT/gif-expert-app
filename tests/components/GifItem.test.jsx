import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {

    const title = 'Gif Title';
    const url = 'https://generic-gif-url.com/cool_image.jpg';

    test('debe de hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={ title } imgUrl={ url }/> );
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ATL indicado', () => {
        render( <GifItem title={ title } imgUrl={ url }/> );

        // This is Ugly -->
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        
        // This is Better :D -->
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });

    test('debe de mostrar el tÃ­tulo en el componente', () => {
        render( <GifItem title={ title } imgUrl={ url }/> );

        expect( screen.getByText(title) ).toBeTruthy();
    });

});
