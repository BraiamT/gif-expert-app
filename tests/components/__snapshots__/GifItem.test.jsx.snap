// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif Title"
      src="https://generic-gif-url.com/cool_image.jpg"
    />
    <p>
      Gif Title
    </p>
  </div>
</div>
`;
